### PR TITLE
GT-96 Django monitoring app

### PR DESCRIPTION
**Pull request by Letizia97**
Features:
- improvements for Python 3 compatibility
- sync table
- sync table with submithosts

### CODE REVIEW

**Comment by tofu-rocketry** [comment]
Can you fix the 'style' category issues that Codeclimate has raised? Don't worry about the other categories of issue for now though, other than the 'clarity' category about the `if` statement.

For line length, follow the [PEP 8](https://peps.python.org/pep-0008/) guidance for longer lines (I need to change the linting to use this setting though):

> Some teams strongly prefer a longer line length. For code maintained exclusively or primarily by a team that can reach agreement on this issue, it is okay to increase the line length limit up to 99 characters, provided that comments and docstrings are still wrapped at 72 characters.

File: monitoring/publishing/models.py [hunk 1]
```diff
@@ -6,28 +6,60 @@
 
 class GridSite(models.Model):
     fetched = models.DateTimeField(auto_now=True)
-    name = models.CharField(max_length=255, primary_key=True)
+    SiteName = models.CharField(max_length=255, primary_key=True)
```
Comment (gregcorbett) [inline monitoring/publishing/models.py:9]: Is it a django thing that these kinds of variables need to be CamelCase?

File: monitoring/publishing/serializers.py [hunk 2]
```diff
@@ -11,7 +16,36 @@ class GridSiteSerializer(serializers.HyperlinkedModelSerializer):
...
+            'RecordCountPublished',
+            'RecordCountInDb',
+            'SyncStatus'
+        )
+
+        # Sitename substitutes pk
+        lookup_field = 'SiteName'
+        extra_kwargs = {
+            'url': {'lookup_field': 'SiteName'}
+        }
```
Comment (gregcorbett) [inline monitoring/publishing/serializers.py:48]: what does this do? Are both `lookup_field` and `extra_kwargs` needed?

**Comment by gregcorbett** [comment]
Looks good, a few questions - see above. I can see codeclimate raised 47 issues, but can't see what they are 😢

File: monitoring/publishing/models.py (hunk 1 above)
Comment (tofu-rocketry) [reply monitoring/publishing/models.py:9]: The name of the field needs to match the relevant database, but this only really applies to the unmanaged models that are designed to be used to pull from the APEL Repository dbs. Probably not needed here (though if we revert we'd have to change any other references).

Comment (tofu-rocketry) [reply monitoring/publishing/models.py:9]: Tracking in #19

**Review by tofu-rocketry — APPROVED** [review]

**Pull request merged**